src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java

```diff
@@ -74,48 +80,71 @@ public CaseDataValidator(final List<FieldValidator> validators) {
             return validate(
                 JacksonUtils.convertValue(dataValue),
                 caseFieldDefinition.getFieldTypeDefinition().getComplexFields(),
-                fieldIdPrefix + dataFieldId + FIELD_SEPARATOR);
+                fieldIdPrefix + dataFieldId + FIELD_SEPARATOR, validationContext
+            );
         } else if (BaseType.get(COLLECTION) == fieldType) {
             final List<ValidationResult> validationResults =
-                validateSimpleField(dataFieldId, dataValue, caseFieldDefinition, fieldIdPrefix, fieldType);
+                validateSimpleField(
+                    dataFieldId, dataValue,
+                    caseFieldDefinition,
+                    fieldIdPrefix,
+                    fieldType,
+                    validationContext
```

Comment:
what's the point of having the ValidationContext object if we've left this method as is with all the parameters?
the goal of introducing a ValidationContext object is so that you can simplify to:
`
validateSimpleField(validationContext)
`

Reply:
This is the old code that I did not want to refactor, later in validateSimpleField()  we set all values in the validationContext instance.  I can set the values in validationContext  before calling validateSimpleField(), refactor validateSimpleField() to be like that  validateSimpleField(validationContext) .